File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int>LIS(nums.size(),1);
        for(int i=1; i<LIS.size(); i++) {
            for(int j=0; j<i; j++) {
                if(nums[j]<nums[i]) {
                    LIS[i] = LIS[j]+1 > LIS[i] ? LIS[j]+1 : LIS[i];
                }
            }
        }
        return *max_element(LIS.begin(),LIS.end());
    }
};

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        
        n = len(nums)
        dp = [1] * n
        ans = 1
        
        for i in range(n):
            for j in range(i):
                if nums[j] < nums[i]:
                    dp[i] = max(dp[i],dp[j]+1)
            
            ans = max(ans,dp[i])
        
        return ans